### PR TITLE
CON-1082: Publish to new public Artifactory

### PR DESCRIPTION
**Pull request by filipesoliveira**
There are now two public repositories named conclave-maven-dev for SNAPSHOTs and RCs, and conclave-maven-release for GA.
This commit updates the repositories used by Gradle to publish the artifacts.
Updated the POM files to reference the license used by each component.

### CODE REVIEW

**Comment by filipesoliveira** [comment]
This PR replaces https://github.com/R3Conclave/graal-patches/pull/4.

File: versions.gradle [hunk 1]
```diff
@@ -1,12 +1,5 @@
...
-    // docker container. Besides that, the Graal version used by the Graal module and Conclave must remain separated because
-    // when Graal is upgraded, the graal module must be updated and the graal artifact needs to be published
-    // before updating the graal version used by Conclave. If the Graal versions used by the graal module and Conclave are
-    // updated at the same time, the PR checks for Conclave might fail because the artifact published by the graal module
-    // might not be available on Artifact.
     graal_version = '22.0.0.2'
     graal_commit_id = 'e7d7572e8'
-    conclave_version = '1.3-RC6-SNAPSHOT'
+    conclave_graal_version = '22.0.0.2-1.3-RC6-SNAPSHOT'
 }
```
Comment (bon000) [inline versions.gradle:6]: Sorry for being pedantic, is there anything preventing us from having
```
conclave_graal_version = "$graal_version-1.3-RC6-SNAPSHOT"
```
or similar? (I haven't double checked, just asking).

File: scripts/devenv_envs.sh [hunk 2]
```diff
@@ -4,7 +4,5 @@ set -euo pipefail
 export OBLIVIUM_CONTAINER_REGISTRY_URL=${OBLIVIUM_CONTAINER_REGISTRY_URL:-"localhost:5000"}
 export OBLIVIUM_CONTAINER_REGISTRY_USERNAME=${OBLIVIUM_CONTAINER_REGISTRY_USERNAME:-}
 export OBLIVIUM_CONTAINER_REGISTRY_PASSWORD=${OBLIVIUM_CONTAINER_REGISTRY_PASSWORD:-}
-export OBLIVIUM_MAVEN_URL=${OBLIVIUM_MAVEN_URL:-"https://software.r3.com/artifactory"}
-export OBLIVIUM_MAVEN_REPOSITORY=${OBLIVIUM_MAVEN_REPOSITORY:-"sgxjvm-maven-nightly"}
 export OBLIVIUM_MAVEN_USERNAME=${OBLIVIUM_MAVEN_USERNAME:-}
 export OBLIVIUM_MAVEN_PASSWORD=${OBLIVIUM_MAVEN_PASSWORD:-}
```
Comment (bon000) [inline scripts/devenv_envs.sh]: I forgot to comment this one before, apologies.
If username and password are only used when publishing, wouldn't be worth to remove those variables as well and have everything under `build.gradle`?

**Review by bon000 — CHANGES_REQUESTED** [review]
A couple of minor points, not a blocker.
Sorry for not having reviewed them before

File: graal-sdk/build.gradle [hunk 3]
```diff
@@ -44,9 +44,18 @@ publishing {
                 // Details about the artifact being published
                 groupId = "com.r3.conclave"
                 artifactId = "graal-sdk"
-                version = "$graal_version-$conclave_version"
+                version = conclave_graal_version
```
Comment (shamsasari) [inline graal-sdk/build.gradle]: In the `allprojects` block, can you set `version` to this value. Then all the sub-modules will have this version, and then automatically the publication will use this value as well.

File: versions.gradle (hunk 1 above)
Comment (filipesoliveira) [reply versions.gradle:6]: Unfortunately, that does not work. That is why I created a function  to calculated conclave_graal_version instead if using a variable.

Comment (bon000) [reply versions.gradle:6]: Even with double quotes? ` " `

Comment (filipesoliveira) [reply versions.gradle:6]: The double quotes does the trick. Thanks Marco.

**Review by bon000 — APPROVED** [review]
This looks very ok to me, many thanks.
I guess we will test it shortly as I am working on the PR which is linked to this.

**Comment by filipesoliveira** [comment]
@shamsasari, can this PR be merged?

File: build.gradle [hunk 4]
```diff
@@ -94,6 +92,14 @@ allprojects {
                 name = "build"
                 url = "$buildDir/repo"
             }
+
+            // Artifacts properties - Set the properties that should be used by all artifacts
+            publications {
+                maven(MavenPublication) {
+                    groupId = "com.r3.conclave"
+                    version = conclave_graal_version
```
Comment (shamsasari) [inline build.gradle:100]: Can you not set these inside `allprojects` block above? We do this in common.gradle in the SDK repo. By doing this don't need this `publications` block.

**Comment by shamsasari** [comment]
I have one more question.

File: build.gradle (hunk 4 above)
Comment (filipesoliveira) [reply build.gradle:100]: I am not sure if I am following. This is the graal-patches project. Don't we need to set the groupId and version for the Maven coordinates? Besides this is a comment from above:
image

Comment (shamsasari) [reply build.gradle:100]: Let me raise a PR instead.

**Pull request merged**